db/1.9/modules/ROOT/pages/database-connector-select.adoc

```diff
@@ -0,0 +1,184 @@
+image::database-select-operation-2.png[Input Parameters configuration in Studio]
+
+In the XML editor, the `<db:input-parameters>` configuration looks like this:
+[source,xml,linenums]
+----
+<flow name="DatabaseFlow">
+  <db:select config-ref="dbConfig">
+    <db:sql>SELECT * FROM PLANET WHERE name = :name</db:sql>
+    <db:input-parameters>
+      #[{'name' : payload}]
+    </db:input-parameters>
+  </db:select>
+</flow>
+----
+
+[NOTE]
+====
+*Since version 1.4.0 (escaping colons)*:
+If you need to use the colon character (`:`) in your SQL query, you can escape it
+by putting a backslash before it. This is useful when using PostgreSQL type
+casting, which requires two colons before the type you are casting to, for example:
+
+`<db:sql>SELECT price\:\:float8 FROM PRODUCT</db:sql>`
+====
+
+[NOTE]
+====
+*Since version 1.8.0 (SQL Casting for PostgreSQL and Snowflake)*:
+Database Connector now accepts SQL Casting PostgreSQL and Snowflake syntax's, through the double colon (::) expression without the need to escape each colon (this feature doesn't affect the escaping colons behaviour). For example:
+
+`<db:sql>SELECT MAX(modified_date)::DATE FROM sales</db:sql>`
+====
+
+
+== Configure Dynamic Queries
+
+When you need to parameterize not only the `WHERE` clause, but also parts of the query itself, for example queries that need to compare tables that depend on a condition, or complex queries for which the project table columns need to vary, you can configure dynamic queries.
+
+In the following example, you configure a dynamic query by using a full expression with a string in which the table depends on a variable `$(vars.table)`. Although, some of the query text is dynamic (`"SELECT * FROM $(vars.table)`), the `WHERE` clause still defines the `WHERE` condition using input parameters, in this case, `WHERE name = :name`.
+
+. In your Studio flow, select the *Select* operation.
+. In the operation configuration screen, set the *SQL Query Text* field to `SELECT * FROM $(vars.table) WHERE name = :name`.
+. Set the *Input Parameters* field to `{'name' : payload}`.
+
+The following screenshot shows the dynamic query configuration:
+
+.Dynamic query configuration
+image::database-select-operation-3.png[Dynamic query configuration configuration in Studio]
+
+In the XML editor, the `<db:sql>` configuration looks like this:
+
+[source,xml,linenums]
+----
+<set-variable variableName="table" value="PLANET"/>
+<db:select config-ref="dbConfig">
+    <db:sql>#["SELECT * FROM $(vars.table) WHERE name = :name"]</db:sql>
+    <db:input-parameters>
+        #[{'name' : payload}]
+    </db:input-parameters>
+</db:select>
+----
+
+Input parameters can be applied only to parameters in a `WHERE` clause. To modify any other part of the query, use the DataWeave’s interpolation operator.
+
+
+== Stream Large Results
+
+Use streaming with queries that return many records, such as in integration use cases. In Mule 4, streaming is transparent and always enabled.
```

Comment:
Can we preceded this with the reason? For example, "For better performance, use...."?